common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -477,7 +454,7 @@ private boolean sendReportWorker(boolean explicitRequest) {
             }
 
             // TODO should probably prepend and do it in bulk
-            for (SpanRecord span : req.span_records) {
+            for (Span span : reqBuilder.getSpansList()) {
```

Comment:
btw, this recovery path is debatable... it's also defensible to simply drop the data as long as we increment the client-dropped counter and report that information back to the collectors eventually.